### PR TITLE
Use extern parameter for variable substitution

### LINKED ISSUE #11698: Send variables as options to the query endpoint when submitting a Flux script
When a user submits a Flux script in the data explorer or view editor overlay, the definitions for all template variables used in the script should be added to an object called `options` that is passed to the query endpoint.

### PR DESCRIPTION
**Pull request by chnn**
Closes #11698 

This PR changes our query execution code paths to use the new [`extern`](https://github.com/influxdata/influxdb/commit/af258e62e77e6ca45255de87cd8e57222e2c82b0#diff-33aa613ab26f77a960b47c92852150af) parameter to perform variable substitution. This in preparation for #12162

I ran into a few code quality issues in the app that I would like to fix up:

https://github.com/influxdata/influxdb/issues/12264
https://github.com/influxdata/influxdb/issues/12266

As well as a bug:

https://github.com/influxdata/influxdb/issues/12265

### CODE REVIEW

File: ui/ast.json [hunk 1]
```diff
@@ -0,0 +1 @@
...xpression","location":{"start":{"line":2,"column":12},"end":{"line":2,"column":36},"source":"start: -10s, stop: now()"},"properties":[{"type":"Property","location":{"start":{"line":2,"column":12},"end":{"line":2,"column":23},"source":"start: -10s"},"key":{"type":"Identifier","location":{"start":{"line":2,"column":12},"end":{"line":2,"column":17},"source":"start"},"name":"start"},"value":{"type":"UnaryExpression","location":{"start":{"line":2,"column":19},"end":{"line":2,"column":23},"source":"-10s"},"operator":"-","argument":{"type":"DurationLiteral","location":{"start":{"line":2,"column":20},"end":{"line":2,"column":23},"source":"10s"},"values":[{"magnitude":10,"unit":"s"}]}}},{"type":"Property","location":{"start":{"line":2,"column":25},"end":{"line":2,"column":36},"source":"stop: now()"},"key":{"type":"Identifier","location":{"start":{"line":2,"column":25},"end":{"line":2,"column":29},"source":"stop"},"name":"stop"},"value":{"type":"CallExpression","location":{"start":{"line":2,"column":31},"end":{"line":2,"column":36},"source":"now()"},"callee":{"type":"Identifier","location":{"start":{"line":2,"column":31},"end":{"line":2,"column":34},"source":"now"},"name":"now"}}}]}]}}}]}]}}
```
Comment (chnn) [inline ui/ast.json]: I'll delete this, it shouldn't be checked in.

File: ui/src/timeMachine/utils/queryBuilder.test.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ describe('buildQuery', () => {
     }
 
     const expected = `from(bucket: "b0")
-  |> range(start: timeRangeStart, stop: timeRangeStop)
+  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
```
Comment (OfTheDelmer) [inline ui/src/timeMachine/utils/queryBuilder.test.ts]: Nice

File: ui/src/variables/utils/formatVarsOption.test.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    ]
+
+    const actual = formatVarsOption(variables)
+
+    expect(actual).toEqual(
+      `option v = {
+  a: -99mo1ns,
+  b: 2010-01-01T00:00:00Z - 99mo1ns,
+  c: now(),
+  d: "howdy",
```
Comment (OfTheDelmer) [inline ui/src/variables/utils/formatVarsOption.test.ts]: Nice

File: ui/src/variables/utils/getTimeRangeVars.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        },
+      },
+    }
+  }
+
+  return [startValue, stopValue]
+}
+
+const isDate = (ambiguousString: string): boolean =>
+  !isNaN(Date.parse(ambiguousString))
```
Comment (OfTheDelmer) [inline ui/src/variables/utils/getTimeRangeVars.ts]: haha

File: ui/src/shared/apis/query.ts [hunk 5]
```diff
@@ -141,3 +144,53 @@ export const executeQuery = (
...
+
+    const extern = buildVarsOption([...variables, ...windowVars])
+    const pendingResult = executeQuery(url, orgID, query, extern)
+
+    cancelExecution = pendingResult.cancel
+
+    return pendingResult.promise
+  })
+
+  return {promise, cancel}
```
Comment (OfTheDelmer) [inline ui/src/shared/apis/query.ts]: would be nice to revisit cancelation patterns one of these days

Comment (chnn) [reply ui/src/shared/apis/query.ts]: Totally. I recently learned about the [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) API, which seems like a way to cancel `fetch` requests. Could be cool to look into.

**Pull request merged**